unified-codes/libs/data/src/lib/types/JWT.ts

```diff
@@ -57,22 +57,23 @@ export class JWT {
     }
   }
 
-  static parseAuthorisation(authorisation: string): JWTToken {
+  static parseRequest(request: any): JWTToken {
+    const { headers } = request;
+    const { authorisation } = headers;
     const [_, jwt] = authorisation.split(' ');
     if (!jwt) throw new JsonWebTokenError('authorisation header malformed');
     const token = new JWTToken(jwt);
     if (!JWT.validateToken(token)) throw new JsonWebTokenError('jwt malformed');
     return token;
   }
 
-  static parseRequest(request: any) {
-    const { authorisation = '' } = request;
-    return JWT.parseAuthorisation(authorisation);
-  }
-
-  static parseResponse(response: any) {
-    const { authorisation = '' } = response;
-    return JWT.parseAuthorisation(authorisation);
+  static async parseResponse(response: any): Promise<JWTToken> {
+    const json = await response.json();
+    const { access_token: jwt } = json;
+    if (!jwt) throw new JsonWebTokenError('access_token malformed');
+    const token = new JWTToken(jwt);
+    if (!JWT.validateToken(token)) throw new JsonWebTokenError('jwt malformed');
+    return token;
```

Comment:
:nit It would be nice to lift these two lines out to a shared method

Reply:
Thought that at first too, but thought it might be a bit overkill at this stage? If the pattern comes up again, then definitely agree it should be extracted (rule of three, anyone?).

Reply:
is the term `underkill` ever used? 🤔 
all good - leave it as you are.

Reply:
Hm, would `underkill` be akin to non-working software or really badly written software? Would it be correct to say @wlthomson has applied lethal force to the problem? I'm not really adding anything here, other than expressing agreement with the rule of three 😸